### ISSUE TITLE
Enclave should have a mechanism to back off rather than spin if ringbuffer is empty

### ISSUE THREAD

**Issue by achamayou** [microsoft/CCF#798]
From ds/messaging.h

```
          // TODO(#performance): If this is ever idle (the underlying
          // buffer has no pending messages and there are no tasks to run), it
          // will spin here. With low traffic this may even be where most
          // execution time is spent. To avoid too many wasteful spins, we
          // should add customisable backoff behaviour here. On the host this
          // could simply be a sleep or a wait on a condition variable, while
          // waiting inside the enclave is more complicated (may initially spin
          // inside, then after some attempts OCALL to sleep outside).
```

**Comment by eddyashton** [comment]
Initial plumbing for this went in #1100. We still want to make this backoff configurable, probably adaptive, and it looks like its currently not even as long as we thought it was.

**Comment by eddyashton** [comment]
This sleep behaviour has been broken for a while, as described and fixed in #1625.
